compiler/fir/fir-deserialization/src/org/jetbrains/kotlin/fir/deserialization/ClassDeserialization.kt

```diff
@@ -88,9 +88,10 @@ fun deserializeClassToSymbol(
             containerSource
         )
     if (status.isCompanion) {
-        parentContext?.let {
-            context.annotationDeserializer.inheritAnnotationInfo(it.annotationDeserializer)
+        require(parentContext != null) {
+            "companion object $classId deserialized without parent context - annotations for moved members will be incorrect"
```

Comment:
Sorry but one of tests from `FirTypeEnhancementTestGenerated` crashes with this exception. Please look at this test (`testIntrinsicCompanionObject`).

Reply:
Hm, I don't really know if this is correct for built-ins, so I've removed the exception